peps/pep-0771.rst

```diff
@@ -350,11 +440,258 @@ This could be achieved with e.g:
+  1.0 prior to using default extras. Suppose that package now defines
+  ``minimal`` in 2.0, then downstream users and packages that want to depend on
+  a minimal version of the package cannot declare the following dependency::
+
+    packaage[minimal]>=1.0
+
+  because ``package[minimal]==1.0`` does not exist (in practice, pip ignores
+  unknown extras, so it might be possible to do this, but there is no guarantee
+  that other tools won't error on an unrecognized extra).
+
+  The easiest solution to this problem is for package authors to define a no-op
+  ``minimal`` extra as soon as possible, even if only planning to adopt default
+  extras further down the road, as it will allow ``package[minimal]`` to work
+  for versions prior to when defaults were adopted.
+
+Avoiding adding too many default dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+One temptation for authors might be to include many dependencies by default since
+they can provide a way to opt out from these. We recommend however that authors
+carefully consider what is included by default to avoid unecessarily bloating
+installations and complicating dependency trees. Using default extras does not
+mean that all extras need to be defaults, and there is still scope for users to
+explicitly opt in to non-default extras.
+
+Inheriting from default extras
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If package authors choose to make an extra be installed by default, it is important
+that they are aware that if users explicitly specify another extra, the default may
+not be installed, unless they use the approach described in `Supporting minimal
+installations while not always removing default extras`_.
+
+There are cases, such as the interchangeable backends, or the ``minimal`` extras,
+where ignoring the default if an extra is explicitly specified is the right
+thing to do. However, for other cases, such as using default extras to include
+recommended dependencies while still providing a way to do minimal installs, it
+may be that many of the other extras *should* explicitly 'inherit' the default
+extra(s), so package authors should carefully consider in which cases they want
+the default extras to be installed.
+
+Incompatible extras
+^^^^^^^^^^^^^^^^^^^
+
+In some cases, it may be that packages have extras that are mutually
+incompatible. In this case, we recommend against making using the default extra
+feature for any extra that contains a dependency that could be incompatible with
+another.
+
+Consider a package that has extras ``package[A]`` and ``package[B]``. Users
+could already currently try and install ``package[A]`` and ``package[B]`` or
+``package[A,B]`` which would result in a broken installation, however it would
+at least be explicit that both extras were being installed. Making ``A`` into a
+default extra however could lead to unintuitive issues. A user could do:
+
+.. code-block:: console
+
+    $ pip install package
```

Comment:
Perhaps add a comment  after the command to specify what is being installed (A as a default)

Reply:
Done